services/blockstorage/test/block_sync_stress_deadlock_test.go

```diff
@@ -23,8 +23,8 @@ import (
 // BlockSync calls ConsensusAlgo.HandleBlockConsensus() when sync wakes up.
 // ConsensusAlgo calls BlockStorage.CommitBlock() when a new block is closed.
 func TestSyncPetitioner_Stress_SingleThreadedConsensusAlgoDoesNotDeadlock(t *testing.T) {
-	test.WithContextWithTimeout(15*time.Second, func(ctx context.Context) {
-		harness := newBlockStorageHarness(t).withSyncNoCommitTimeout(time.Nanosecond)
+	test.WithConcurrencyHarness(t, func(ctx context.Context, parent *test.ConcurrencyHarness) {
```

Comment:
`parent` here is not very clear. the idea that test harnesses are hierarchal is unnecessarily complex.
 
Whatever it is that this bit provides should be referred to in the name because there is no hierarchy of harnesses. if this is the supervision-mixin lets call it that. but this provides functionality rather than serve as the top of a pyramid.

Reply:
More on names: 
If "WithConcurrencyHarness" was called "WithSupervision" or "WithShutdownableHarness" it would be easier to understand. The reason IMHO we run into obscure term ConcurrencyHarness is that we violate the single responsibility principle. i would rather see logger as a third parameter on the test callback signature so we can have a single responsibility to the supervising harness